### PR TITLE
Stylish, hlint

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

#4 

# Overview

- [x] I have copied `.stylish-haskell.yaml`, `.hlint.yaml` from the old repo

# Comments

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+.ghci
+
+### Cardano ###
+
+test_keystore.*
+state-staging
+state-integration
+secret.key
+secret.key.lock
+wallet-db-acid/
```
Comment (Anviking) [inline .gitignore]: Maybe start with a clean .gitignore? Or keep this?

Comment (akegalj) [reply .gitignore]: I would start with a clean one and as someone will be modelling db he/she will add a line to .gitignore before commiting 

EDIT: I would remove "### Cardano ###" part first. When project is in place and when someone launches the whole system (with node and everything) then that person will add exceptions to gitignore

File: .stylish-haskell.yaml [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+  #     # Number of spaces to use for each tab. Default: 8, as specified by the
+  #     # Haskell report.
+  #     spaces: 8
+
+  # Remove trailing whitespace
+  - trailing_whitespace: {}
+
+# A common setting is the number of columns (parts of) code will be wrapped
+# to. Different steps take this into account. Default: 80.
+columns: 80
```
Comment (akegalj) [inline .stylish-haskell.yaml]: note the discussion at https://input-output-rnd.slack.com/archives/GBT05825V/p1551444614041200

File: .gitignore (hunk 1 above)
Comment (KtorZ) [reply .gitignore]: I'd suggest to start with a clean one taken from: https://www.gitignore.io/

File: .hlint.yaml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+# - ignore: {name: Use let}
+# - ignore: {name: Use const, within: SpecialModule} # Only within certain modules
+
+
+# Define some custom infix operators
+# - fixity: infixr 3 ~^#^~
+
+
+# To generate a suitable file for HLint do:
+# $ hlint --default > .hlint.yaml
```
Comment (KtorZ) [inline .hlint.yaml:82]: That's also a big topic of discussion actually :/ ... 
I'd suggest that for now, we do start with the _default_ hlint configuration, and, iterate over the weeks to add or remove stuff we like or don't.

File: .stylish-haskell.yaml [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+  - PolyKinds
+  - QuasiQuotes
+  - RecordWildCards
+  - ScopedTypeVariables
+  - StandaloneDeriving
+  - TemplateHaskell
+  - TupleSections
+  - TypeApplications
+  - TypeFamilies
+  - ViewPatterns
```
Comment (KtorZ) [inline .stylish-haskell.yaml]: Let's run a poll about that one as well, just to decide what rules to put in there.

**Comment by paweljakubas** [comment]
btw. we agreed on using `.editorconfig` How it is going to live along with above files? Don't these files overlap somehow? (maybe stupid question -> I have not used EditorConfig before)

**Comment by KtorZ** [comment]
@paweljakubas No they don't conflicts but they are complementary to each others.

File: .stylish-haskell.yaml [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+# preferred format here.
+#
+# - native: Native newline format. CRLF on Windows, LF on other OSes.
+#
+# - lf: Convert to LF ("\n").
+#
+# - crlf: Convert to CRLF ("\r\n").
+#
+# Default: native.
+newline: native
```
Comment (Anviking) [inline .stylish-haskell.yaml]: Wait, won't this actually interfere with `.editorconfig`? 🤔 

https://github.com/input-output-hk/cardano-wallet-legacy/blob/develop/.editorconfig

- line-endings
- columns

Ok, line-endings in editorconfig is only for the guide, and in stylish it's for wrapping the imports, I think.

And for newlines they can interfere.

File: .stylish-haskell.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Stylish-haskell configuration file
+#
+# See `stylish-haskell --defaults` or
+# https://github.com/jaspervdj/stylish-haskell/blob/master/data/stylish-haskell.yaml
```
Comment (Anviking) [inline .stylish-haskell.yaml:4]: Keeping the comments in this file just got complicated…

**Review by KtorZ — APPROVED** [review]

**Review by KtorZ — APPROVED** [review]

**Pull request merged**